src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/VSTelemetryServiceUtility.cs

```diff
@@ -66,33 +68,43 @@ public static class VSTelemetryServiceUtility
 
         public static async Task<ProjectTelemetryEvent> GetProjectTelemetryEventAsync(NuGetProject nuGetProject)
         {
-            // Get the project details.
-            string projectUniqueName = nuGetProject.GetMetadata<string>(NuGetProjectMetadataKeys.UniqueName);
+            if (nuGetProject == null)
+            {
+                throw new ArgumentNullException(nameof(nuGetProject));
+            }
+            string projectUniqueName = "";
+            ProjectTelemetryEvent returnValue = null;
 
-            // Emit the project information.
             try
             {
+                // Get the project details.
+                projectUniqueName = nuGetProject.GetMetadata<string>(NuGetProjectMetadataKeys.UniqueName);
+
                 string projectId = nuGetProject.GetMetadata<string>(NuGetProjectMetadataKeys.ProjectId);
                 NuGetProjectType projectType = GetProjectType(nuGetProject);
                 bool isUpgradable = await NuGetProjectUpgradeUtility.IsNuGetProjectUpgradeableAsync(nuGetProject);
+                string fullPath = nuGetProject.GetMetadata<string>(NuGetProjectMetadataKeys.FullPath);
 
-                return new ProjectTelemetryEvent(
+                returnValue = new ProjectTelemetryEvent(
                     NuGetVersion.Value,
                     projectId,
                     projectType,
-                    isUpgradable);
+                    isUpgradable,
+                    fullPath);
             }
             catch (Exception ex)
             {
-                string message =
-                    $"Failed to emit project information for project '{projectUniqueName}'. Exception:" +
-                    Environment.NewLine +
-                    ex.ToString();
+                // ArgumentException means project metadata is empty
+                // DTE exceptions could mean VS process has a severe failure
+                string message = string.Format(CultureInfo.CurrentCulture, Strings.Warning_GetProjectTelemetryEventAsync, projectUniqueName, ex);
```

Comment:
ActivityLog messages are not user facing. They are intended for the developer primarily and as such should not be localized.

Reply:
We can keep our activity log messages in english and keep the string in the resources file for code organization and code readability improvements.

What do you think?

```suggestion
                string message = string.Format(CultureInfo.InvariantCulture, Strings.Warning_GetProjectTelemetryEventAsync, projectUniqueName, ex);
```

Reply:
I prefer not doing that simply because the bar for the strings as resources is different. It does affect readability. 

I'm curious what others think though. 

I mostly just care about not localizing it, cause activity log in a language most of the engineers don't understand is not helpful.

Reply:
Reverted to original string message in favor of readability.

Reply:
@dominoFire  FYI that proposed change wouldn't get the English string. Localized strings are obtained in `Strings.Designer.cs`, so by the time this method's `string.Format` is called, it's too late.

Reply:
Which part of VS window is `activity log`? Is it Package Manager restore log?
![image](https://user-images.githubusercontent.com/8766776/108576095-276b3200-72d1-11eb-8024-e4d3954aa4ae.png)

Reply:
No, that's the output window, https://docs.microsoft.com/en-us/visualstudio/extensibility/how-to-use-the-activity-log?view=vs-2019